### PR TITLE
[5.2] enhance dirty check in eloquent model

### PR DESCRIPTION
**Pull request by marco709394**
The modification brings up the efficiency of isDirty check by removing dependency on getDirty, better when there are multiple dirty properties.

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Is there sufficient test coverage to backup that this change is correct?

**Comment by marco709394** [comment]
If talking about the test cases, it is sufficient as the functionality of isDirty is not extended, so DatabaseEloquentModelTest::testDirtyAttributes has covered what it was.

File: src/Illuminate/Database/Eloquent/Model.php [hunk 1]
```diff
@@ -3177,19 +3177,19 @@ public function syncOriginalAttribute($attribute)
...
         }
 
-        foreach ($attributes as $attribute) {
-            if (array_key_exists($attribute, $dirty)) {
-                return true;
+        foreach ($attributes as $key) {
+            if (isset($this->attributes[$key])) {
+                if (! array_key_exists($key, $this->original) ||
+                        $this->attributes[$key] !== $this->original[$key] &&
+                            ! $this->originalIsNumericallyEquivalent($key)) {
```
Comment (taylorotwell) [inline src/Illuminate/Database/Eloquent/Model.php]: This condition is quite hard to read. Is it intended that there is no logical grouping... You have `foo || bar && baz` with no grouping.

**Comment by marco709394** [comment]
Is it better now?

File: src/Illuminate/Database/Eloquent/Model.php [hunk 2]
```diff
@@ -3177,19 +3177,22 @@ public function syncOriginalAttribute($attribute)
...
+            $attributes = array_keys($this->attributes);
+        } elseif (! is_array($attributes)) {
             $attributes = func_get_args();
         }
 
-        foreach ($attributes as $attribute) {
-            if (array_key_exists($attribute, $dirty)) {
-                return true;
+        foreach ($attributes as $key) {
+            if (isset($this->attributes[$key])) {
```
Comment (ArjanSchouten) [inline src/Illuminate/Database/Eloquent/Model.php]: Here you are checking with ```isset``` so the key must exists and the value must not be ```null```. The original check is different to this. Intentional?

**Comment by taylorotwell** [comment]
Holding off on this for now.

**Pull request closed without merging**